lib/publiccloud/instance.pm

```diff
@@ -203,10 +203,57 @@ sub wait_for_guestregister
         }
         sleep 1;
     }
+
     $self->upload_log('/var/log/cloudregister', log_name => $autotest::current_test->{name} . '-cloudregister.log');
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 check_guestregister
+
+    check_guestregister();
+
+Check that the registration state of newly created guest match our requirements.
+https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt
+=cut
+sub check_guestregister
+{
+    my ($self) = @_;
+
+    if (is_byos) {
+        $self->run_ssh_command(cmd => '! sudo zypper lr', timeout => 6000);
+
+        if ($self->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1, quiet => 1) !~ /disabled/) {
```

Comment:
looks like something which needs `validate_ssh_command_output` ? if you know what I mean ?

Reply:
I don't. I would first of all split and refactor `run_ssh_command`.

Reply:
I mean something like this https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1245 but instead of `script_output` call you doing `run_ssh_command` . And what you have on your mind when you saying about refactor `run_ssh_command` ?

Reply:
It's probably for separate discussion. Do you think that I should do this in this pull request?

Reply:
+1 , this is just thoughts for the future . resolving the thread